lib/fog/core/services_mixin.rb

```diff
@@ -31,9 +31,14 @@ def require_service_provider_library(service, provider)
     end
 
     def service_provider_constant(service_name, provider_name)
-      Fog.const_get(service_name).const_get(provider_name, false)
+      Fog.const_get(service_name).const_get(*const_get_args(provider_name))
     rescue NameError  # Try to find the constant from in an alternate location
-      Fog.const_get(provider_name).const_get(service_name, false)
+      Fog.const_get(provider_name).const_get(*const_get_args(service_name))
+    end
+
+    def const_get_args(*args)
+      args += [false] unless RUBY_VERSION < '1.9'
+      args
```

Comment:
Also, didn't we deprecate Ruby 1.8.7/1.9.3 across the board?
If not 1.9.3 then 1.8.7 for sure I thought...